libpod/runtime_pod.go

```diff
@@ -129,3 +130,59 @@ func (r *Runtime) GetLatestPod() (*Pod, error) {
 	}
 	return pods[lastCreatedIndex], nil
 }
+
+// GetRunningPods returns an array of running pods
+func (r *Runtime) GetRunningPods() ([]*Pod, error) {
+	var (
+		cids        []string
+		runningPods []*Pod
+	)
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	if !r.valid {
+		return nil, ErrRuntimeStopped
+	}
+	pods, err := r.state.AllPods()
+	if err != nil {
+		return nil, err
+	}
+	containers, err := r.GetRunningContainers()
+	if err != nil {
+		return nil, err
+	}
+	// Assemble a slice of cids for lookup
+	for _, c := range containers {
+		cids = append(cids, c.ID())
+	}
+	for _, p := range pods {
+		containersByID, err := p.AllContainersByID()
```

Comment:
No need for this - go through all the containers from `GetRunningContainers` and build a set of `c.Pod()`, then get all of those pods

Reply:
why is that better

Reply:
You skip all the expensive `AllContainersByID` calls

Reply:
fixed